app/controllers/comment.js

```diff
@@ -0,0 +1,131 @@
+const CommentModel = require('../models/Comment')
+
+module.exports = {
+  comment: async (req, res, next) => {
+    const comment = new CommentModel(req.body)
+    try {
+      await comment.save()
+      res.status(201).json({ comment: comment })
+    } catch (error) {
+      console.log(error)
+      res.status(400).json({ error: error })
+    }
+  },
+  delete: async (req, res, next) => {
+    const { id } = req.params
+    console.log(id)
+    try {
+      const comment = await CommentModel.findByIdAndRemove(id)
+      if (!comment) {
+        return res.status(400).json({ error: 'No comment exixts' })
+      }
+      res.status(200).json({ comment: comment })
+    } catch (error) {
+      console.log(error)
+      res.status(400).json({ error: error })
+    }
+  },
+  update: async (req, res, next) => {
+    const { id } = req.params
+    const updates = Object.keys(req.body)
+    let isValidUpdate = true
+    if (updates !== 'content') {
+      isValidUpdate = false
+    }
+    if (!isValidUpdate) {
+      return res.status(400).json({ error: 'Wrong Update Request' })
+    }
+    try {
+      const comment = await CommentModel.findById(id)
+      updates.forEach(update => {
+        comment[update] = req.body[update]
+      })
+      await comment.save()
+      res.status(204).json({ comment: comment })
+    } catch (error) {
+      console.log(error)
+      res.status(400).json({ error: error })
+    }
+  },
+  getCommentByPost: async (req, res, next) => {
+    const { id } = req.params
+    try {
```

Comment:
Also populate the useId so that we can render in the client side that who has posted this comment

Reply:
I think this is already done. In the comment schema there is a field which specifies the userID of the person who has posted the comment. Here, the message I am returning is the list of comments which belong to the post from where we can iterate and fetch the userID.

Reply:
@rak-shit Without fetching user by userId we can simpley populate them and retrieve the value and send it to the client side.
Just google a little bit about mongoose populate function.

Reply:
Sounds good. I understood what you said.